pkg/metrics/features/policy.go

```diff
@@ -94,10 +122,16 @@ func ruleType(r api.Rule) RuleFeatures {
 		}
 	}
 
-	if !(rf.allFeaturesEgressCommon()) {
+	if !(rf.allFeaturesEgressCommon() && rf.allFeaturesPortRules()) {
 		for _, e := range r.Egress {
 			ruleTypeEgressCommon(&rf, e.EgressCommonRule)
-			if rf.allFeaturesEgressCommon() {
+			if !rf.allFeaturesPortRules() {
+				if len(e.ToFQDNs) > 0 {
+					rf.DNS = true
+				}
+				ruleTypePortRules(&rf, e.ToPorts)
```

Comment:
So this metric will also count `ToFQDN` rules as well as `DNS` rules? What if the user has only `ToFQDN` rules and not `DNS` or vice versa? A common mistake is to define a `ToFQDN` without a corresponding DNS rule. I'm not sure if we want to expose that at this level though. :thinking:

Reply:
That's why I wanted someone from the policy sig to review this PR as I was not so sure about this as well. Should we then distinguish between DNS and ToFQDN policies?

Reply:
My feeling is yes we should distinguish between them. Otherwise, it can be misleading. How do others in @cilium/sig-policy feel?

Reply:
Done

Reply:
I see that the changes were made in the last commit to add separate counting for FQDN, but don't we need to remove the 

``
if len(e.ToFQDNs) > 0 {
					rf.DNS = true
				}
```

block too?

Reply:
Addressed in https://github.com/cilium/cilium/pull/36203#discussion_r1868327428